src/box/txn.h

```diff
@@ -893,6 +893,38 @@ txn_is_nop(const struct txn *txn)
 	return txn->n_new_rows + txn->n_applier_rows == 0;
 }
 
+static inline bool
+txn_is_fully_temporary(struct txn *txn)
+{
+	if (!txn_is_nop(txn))
+		return false;
+	struct txn_stmt *stmt;
+	stailq_foreach_entry(stmt, &txn->stmts, next) {
+		if (stmt->space != NULL && !space_is_temporary(stmt->space) &&
+		    space_is_data_temporary(stmt->space))
```

Comment:
Nit: hm, that looks weird. Perhaps it would be better to check here literally `space->opts.type == SPACE_TYPE_DATA_TEMPORARY`. Instead of these 2 checks.

Reply:
Fixed.